src/qforte/utils/circuit_to_latex.py

```diff
@@ -0,0 +1,145 @@
+        qubit_ids.add(gate.target())
+        qubit_ids.add(gate.control())
+
+    # initialize wires and gate_count dictionaries
+    for wire in qubit_ids:
+        wires[wire] = "            \lstick{$\ket{q_{"+str(wire)+"}}$}"
+        gate_count[wire] = 0
+
+    for gate in circ.gates():
+        # To help the tex compilation, large quantum circuits are split
+        # into smaller ones. If the maximum number of allowed gates in a
+        # given wire has been reached and a new gate will be added to that
+        # wire, create a new tikz picture
+        current_max_gate_count = max(gate_count.values())
+        if current_max_gate_count == max_gate_count_per_tikz:
+            wires_with_max_gate_count_per_tikz = [wire for wire in gate_count if gate_count[wire] == max_gate_count_per_tikz]
+            if gate.target() in wires_with_max_gate_count_per_tikz or gate.control() in wires_with_max_gate_count_per_tikz:
+                for wire in wires:
+                    for i in range(max_gate_count_per_tikz - gate_count[wire]):
+                        wires[wire] += " & \qw"
+                for wire in reversed(list(qubit_ids)[1:]):
+                    texfile.write(wires[wire]+" \\\ \n")
+                wire = list(qubit_ids)[0]
+                texfile.write(wires[wire]+"\n")
+                texfile.write(tikz_end)
+                texfile.write(tikz_start)
+                for wire in qubit_ids:
+                    wires[wire] = "            \ \ldots \ \qw"
+                    gate_count[wire] = 0
+        # check if gate is acting on single qubit
+        if gate.target() == gate.control():
+            wires[gate.target()] += " & \gate{" + gate.gate_id() + "}"
+            gate_count[gate.target()] += 1
+        else:
+            # currently QForte supports up to two-qubit gates
+            count_target = gate_count[gate.target()]
+            count_control = gate_count[gate.control()]
+            # if the target and control wires contain a different
+            # number of gates, we need to add "\qw" to equalize them
+            if count_target < count_control:
+                for i in range(count_control - count_target):
+                    wires[gate.target()] += " & \qw"
+                    gate_count[gate.target()] += 1
+            elif count_target > count_control:
+                for i in range(count_target - count_control):
+                    wires[gate.control()] += " & \qw"
+                    gate_count[gate.control()] += 1
+            diff = int(gate.control() - gate.target())
+            if gate.gate_id() not in ["cZ", "aCNOT", "acX", "CNOT", "cX"]:
+                raise ValueError("The only two-qubit gates that are currently supported are: CNOT, aCNOT, and cZ!")
+            if gate.gate_id() == "cZ":
+                wires[gate.target()] += " & \\ctrl{}"
+            else:
+                wires[gate.target()] += " & \\targ{}"
+            if gate.gate_id() == 'aCNOT':
```

Comment:
I think I'm missing something. cZ and aCNOT gates are distinct, but how does a reader separate acX and CNOT and cX from each other?

Reply:
The logic is as follows. If we are dealing with a cZ gate, a "control" (`\ctrl{}`) symbol is inserted at the target qubit. Otherwise, a ⊕ symbol (`\targ{}`) is placed at the target qubit. This symbol is common to both aCNOT and CNOT. Subsequently, if we are dealing with an aCNOT, an anti-control (`\octrl{...}`) symbol is placed on the "control" qubit. Otherwise a control {`\ctrl{...}`} symbol is inserted. The difference between the `\octrl` and `ctrl` commands is that one creates an open circle and the other a solid one.